src/js/client/common-event-bindings.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * Common Event Bindings
+ *
+ * Provides a grade `gpii.binder.bindMarkupEvents` to streamline the
+ * process of attaching html events such as `click` to selectors.
+ *
+ * Copyright 2019 Raising the Floor - International
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this License.
+ *
+ * You may obtain a copy of the License at
+ * https://github.com/GPII/universal/blob/master/LICENSE.txt
+ */
+
+"use strict";
+
+/**
+ * GPII Binder Markup Events Grade
+ * This grade allows binding typical HTML events such as
+ * mouse clicks and keypress events to selectors each time
+ * the markup is rendered for a component. (Meaning they will
+ * also apply if a component is rerendered after a model refresh
+ * or similar situation)
+ *
+ * It adds a new area to a grades options called `markupEventBindings`
+ * which allows binding `selectors` to jQuery events. Theoretically
+ * other event constructs could be supported in the future, but only
+ * jQuery events are implemented at the time of writing.
+ *
+ * Example usage of adding a click handler to a selector productListLinks.
+ * ```
+ * markupEventBindings: {
+ *     productListLinks: {
+ *         // type: jQuery <- Defaults to jQuery but could be configured ITF
+ *         method: "click",
+ *         args: ["{that}.selectProduct"]
+ *     }
+ * }
+ * ```
+ */
+fluid.defaults("gpii.binder.bindMarkupEvents", {
```

Comment:
Give links to the various places where this code is now duplicated

Reply:
I've added a comment and link to the Jira ticket tracking this:
https://issues.gpii.net/browse/GPII-2933